### PR TITLE
Fix #109946, fix #109951: Better paper size choices for Page Settings

### PR DESCRIPTION
**Pull request by ZackTheCardshark**
Add dimensions to paper names in Page Settings dropdown; add Concert Part, Flip Folder, Octavo and Hymn sizes; straighten out Folio vs. F4; more common Executive; describe Ledger and Tabloid in inches

### CODE REVIEW

File: libmscore/page.cpp [hunk 1]
```diff
@@ -36,38 +36,43 @@ namespace Ms {
...
-      PaperSize("Ledger",    MM(432),  MM(279)),
-      PaperSize("Tabloid",   MM(279),  MM(432)),
-      PaperSize(0,           MM(1),    MM(1))   // mark end of list
+      PaperSize("Custom",                       MM(1),      MM(1)),
+      PaperSize("A4 (210 x 297 mm)",            MM(210),    MM(297)),
+      PaperSize("B5 (176 x 250 mm)",            MM(176),    MM(250)),
+      PaperSize("Letter (8.5 x 11 in)",         8.5,        11),
+      PaperSize("Legal (8.5 x 14 in)",          8.5,        14),
+      PaperSize("Executive (7.25 x 10.5 in)",   7.25,       10.5),
+      PaperSize("Concert Part (9 x 12 in)"),    9,          12,
```
Comment (Jojo-Schmitz) [inline libmscore/page.cpp]: Here's a `)` too much

File: libmscore/page.cpp [hunk 2]
```diff
@@ -36,38 +36,43 @@ namespace Ms {
...
+      PaperSize("Flip Folder (7 x 5.25 in)",    7,          5.25,
+      PaperSize("Octavo (6.75 x 10.5 in"),      6.75,       10.5,
+      PaperSize("Hymn (5.75 x 8.25)"),          5.75,       8.25
```
Comment (Jojo-Schmitz) [inline libmscore/page.cpp]: here too

File: libmscore/page.cpp [hunk 3]
```diff
@@ -36,38 +36,43 @@ namespace Ms {
...
+      PaperSize("B4 (250 x 353 mm)",            MM(250),    MM(353)),
+      PaperSize("B6 (125 x 176 mm)",            MM(125),    MM(176)),
+      PaperSize("B7 (88 x 125 mm)",             MM(88),     MM(125)),
+      PaperSize("B8 (62 x 88 mm)",              MM(62),     MM(88)),
+      PaperSize("B9 (44 x 62 mm)",              MM(44),     MM(62)),
+      PaperSize("B10 (31 x 44 mm)",             MM(31),     MM(44)),
+      PaperSize("Comm10E (105 x 241 mm)",       MM(105),    MM(241)),
+      PaperSize("DLE (110 x 220 mm)",           MM(110),    MM(220)),
+      PaperSize("Folio (8.5 x 13 in)",          8.5,        13,
+      PaperSize("F4 (210 x 330 mm)",)           MM(210),    MM(330)),
```
Comment (Jojo-Schmitz) [inline libmscore/page.cpp]: and here

File: libmscore/page.cpp [hunk 4]
```diff
@@ -36,38 +36,43 @@ namespace Ms {
...
-      PaperSize(0,           MM(1),    MM(1))   // mark end of list
+      PaperSize("Custom",                       MM(1),      MM(1)),
+      PaperSize("A4 (210 x 297 mm)",            MM(210),    MM(297)),
+      PaperSize("B5 (176 x 250 mm)",            MM(176),    MM(250)),
+      PaperSize("Letter (8.5 x 11 in)",         8.5,        11),
+      PaperSize("Legal (8.5 x 14 in)",          8.5,        14),
+      PaperSize("Executive (7.25 x 10.5 in)",   7.25,       10.5),
+      PaperSize("Concert Part (9 x 12 in)"),    9,          12,
+      PaperSize("Flip Folder (7 x 5.25 in)",    7,          5.25,
+      PaperSize("Octavo (6.75 x 10.5 in"),      6.75,       10.5,
```
Comment (Jojo-Schmitz) [inline libmscore/page.cpp]: here `"` and `)` are swapped

**Comment by lasconic** [comment]
Could you create an issue in the issue tracker, and refer to it in the commit message? In the issue, it would be good to link to a source for the names and dimensions, a wikipedia page is fine, so we can refer to it when a user will come and say "hey, this is nonsense, flip folder are XxY in my country..."

File: libmscore/page.cpp [hunk 5]
```diff
@@ -36,38 +36,43 @@ namespace Ms {
...
-      PaperSize(0,           MM(1),    MM(1))   // mark end of list
+      PaperSize("Custom",                       MM(1),      MM(1)),
+      PaperSize("A4 (210 x 297 mm)",            MM(210),    MM(297)),
+      PaperSize("B5 (176 x 250 mm)",            MM(176),    MM(250)),
+      PaperSize("Letter (8.5 x 11 in)",         8.5,        11),
+      PaperSize("Legal (8.5 x 14 in)",          8.5,        14),
+      PaperSize("Executive (7.25 x 10.5 in)",   7.25,       10.5),
+      PaperSize("Concert Part (9 x 12 in)",     9,          12,
+      PaperSize("Flip Folder (7 x 5.25 in)",    7,          5.25,
+      PaperSize("Octavo (6.75 x 10.5 in)",      6.75,       10.5,
```
Comment (Jojo-Schmitz) [inline libmscore/page.cpp]: Closing brack missing near end of line on the 3 lines above

File: libmscore/page.cpp [hunk 6]
```diff
@@ -36,38 +36,43 @@ namespace Ms {
...
+      PaperSize("Hymn (5.75 x 8.25)",           5.75,       8.25
```
Comment (Jojo-Schmitz) [inline libmscore/page.cpp]: closing bracket and comma missing at end of line

File: libmscore/page.cpp [hunk 7]
```diff
@@ -36,38 +36,43 @@ namespace Ms {
...
+      PaperSize("B3 (353 x 500 mm)",            MM(353),    MM(500)),
+      PaperSize("B4 (250 x 353 mm)",            MM(250),    MM(353)),
+      PaperSize("B6 (125 x 176 mm)",            MM(125),    MM(176)),
+      PaperSize("B7 (88 x 125 mm)",             MM(88),     MM(125)),
+      PaperSize("B8 (62 x 88 mm)",              MM(62),     MM(88)),
+      PaperSize("B9 (44 x 62 mm)",              MM(44),     MM(62)),
+      PaperSize("B10 (31 x 44 mm)",             MM(31),     MM(44)),
+      PaperSize("Comm10E (105 x 241 mm)",       MM(105),    MM(241)),
+      PaperSize("DLE (110 x 220 mm)",           MM(110),    MM(220)),
+      PaperSize("Folio (8.5 x 13 in)",          8.5,        13,
```
Comment (Jojo-Schmitz) [inline libmscore/page.cpp]: closing paren missing near end of line

File: libmscore/page.cpp (hunk 5 above)
Comment (Jojo-Schmitz) [inline libmscore/page.cpp]: Sorry, I missed those last time. Build if you'd be compiling it yourself, you'd had noticed ;-)

**Comment by ZackTheCardshark** [comment]
Thanks for spotting them all, Jojo. I actually compiled to check things a few times along the way to this point, but late last night I made the final changes, opened the PR, and went to bed. ;-)

Nicolas, I'm actually going to open two issues—one for the new page sizes, and one for including the dimensions in the size's name.

**Comment by MarcSabatella** [comment]
FWIW, I applaud this effort!

**Comment by Jojo-Schmitz** [comment]
Hmm, None of these strings are translated, but at least "Custom" should, not too sure about the rest though

**Comment by Jojo-Schmitz** [comment]
If you'll to the Change from `"Custom"` to `QT_TRANSLATE_NOOP("paperSizes", "Custom")` and equivalent for other strings that should get translated, I'll do the rest in a different PR. If I'd do that change too, we'd be running into a conflict.

**Comment by lasconic** [comment]
There will be conflicts anyway with https://github.com/musescore/MuseScore/pull/2573 ...

**Comment by Jojo-Schmitz** [comment]
Adding yet another conflict to the mess doesn't seem like a good option, does it? I've added the changes needed to get the translation enabled to my PR #2579 for now

**Comment by Jojo-Schmitz** [comment]
@ZackTheCardshark: feel free to 'steel' that code...

**Comment by ZackTheCardshark** [comment]
I don't pretend to have any clue what that meant or how it should work, but is this correct?

**Comment by Jojo-Schmitz** [comment]
That would make them all translatable, not sure we want ot need that. A4 for example is A4 in any language, and ist size is also the same all over the planet. "Custom" should be translatable,, IMHO Letter and Legal, all the A* and B* formats should not, and I'm not sure about Concert Part, Flip Folder, Octavo, Hymn. Are these the official names in some standard?

**Comment by Jojo-Schmitz** [comment]
If you need documentation, http://doc.qt.io/qt-5/qpagesize.html has quite a few paper formats, with names and some with sizes. And there is  https://en.wikipedia.org/wiki/Paper_size. The latter for example has "Choral Octavo" where you have "Octavo", and "Marching Band Flip-Folder", where you have "Flip Folder", the latter with a slightly different size, 6,75x5,25 rather than 7x5.25. Haven't found sources for "Concert Part" and "Hymn",

**Comment by ZackTheCardshark** [comment]
It must be admitted the music sizes are not exactly fixed standards.

* Traditional concert parts are generally on 9 x 12 or 9.5 x 12.5 size paper. Sibelius defines both those sizes by default as well as 10 x 13, which I've never seen; Finale only does 9 x 12. But 9 x 12 paper is much more readily available, even if it doesn't have a standard name outside of music.

* I don't think there's an official flip folder paper size, but it's a common enough need to get music into flip folders, and even if exactly sized paper isn't available, it's still important to format it appropriately before printing and cutting the paper to make it fit. The user will still have to manually change to narrower margins and a smaller staff space, unless MuseScore's load/save style controls get much more powerful in 3.0. I'll defer to Wikipedia on this one and make the width less.

* I'm also willing to prepend "Choral" to Octavo, per Wikipedia.

* Hymn size is defined this way in both Finale and Sibelius.

As to the translations, one question that becomes relevant now is whether the abbreviations "mm" and "in" are universal? Because now the name of every page size includes one of those.

**Comment by ZackTheCardshark** [comment]
Forgot to mention: Executive is a confusing one. I think the change I proposed is the one more commonly found in lists of paper sizes on the internet, but perhaps not.

**Comment by ZackTheCardshark** [comment]
Feature request for additional sizes: https://musescore.org/en/node/109946

**Comment by ZackTheCardshark** [comment]
Feature request for including measurements in size names: https://musescore.org/en/node/109951

**Comment by Jojo-Schmitz** [comment]
I wouldn't translate units. Not MMS at least, being an international standard, SI. And while the word inch would get translated (to Zoll in German), in most probably would not. Marking the sized translatable is asking for trouble and mistakes, so better don't. And all the formats defined by some international standard should not get translated IMHO. So I think this leaves Custom,  Concert part, Flip folder, Octavo and Hymn for translation, at most.

**Comment by lasconic** [comment]
So what do we do with this one? 
I tend to agree that asking to translate A4 and B4 and friends is looking for trouble, so I wouldn't translate them. For the other ones, it will be hard to not translate the units... we would need another columns with the unit and a function to concatenate them... not sure it's worth it.

**Comment by ZackTheCardshark** [comment]
So now only "Custom," "Concert Part," "Flip Folder," "Choral Octavo" and "Hymn" are translatable; the rest, including Letter, Executive, etc. are not.

**Comment by lasconic** [comment]
That's good. Check your syntax (maybe compile the code on your computer) around A7.
And if you could use http://doc.qt.io/qt-5/qtglobal.html#QT_TRANSLATE_NOOP3 and add a comment like "Do not translate inches" that would be perfect.

**Comment by ZackTheCardshark** [comment]
Sorry I missed that; I'm away from my home computer, using the online
GitHub editor, so I can't compile at the moment.

Despite appearances, at this point I still don't know anything about code.
How should QT_TRANSLATE_NOOP3 be used in this context?

**Comment by Jojo-Schmitz** [comment]
something like `PaperSize(QT_TRANSLATE_NOOP("paperSizes", "Concert Part") + " (9 x 12 in)"),  9, 12)` maybe?

**Comment by lasconic** [comment]
Use 

```
      //: Do not translate "in".
      PaperSize(QT_TRANSLATE_NOOP("paperSizes", "Concert Part (9 x 12 in)"), 9, 12),
      //: Do not translate "in".
      PaperSize(QT_TRANSLATE_NOOP("paperSizes", "Flip Folder (6.75 x 5.25 in)"),   6.75, 5.25),
      //: Do not translate "in".
      PaperSize(QT_TRANSLATE_NOOP("paperSizes", "Choral Octavo (6.75 x 10.5 in)"), 6.75, 10.5),
      //: Do not translate "in".
      PaperSize(QT_TRANSLATE_NOOP("paperSizes", "Hymn (5.75 x 8.25 in)"), 5.75, 8.25),
```

File: libmscore/page.cpp [hunk 8]
```diff
@@ -36,38 +36,48 @@ namespace Ms {
...
+                6.75,       10.5),
+      PaperSize(QT_TRANSLATE_NOOP("paperSizes","Hymn (5.75 x 8.25 in)"),
+                5.75,       8.25),
+      PaperSize("A0 (841 x 1189 mm)",                      MM(841),    MM(1189)),
+      PaperSize("A1 (594 x 841 mm)",                       MM(594),    MM(841)),
+      PaperSize("A2 (420 x 594 mm)",                       MM(420),    MM(594)),
+      PaperSize("A3 (297 x 420 mm)",                       MM(297),    MM(420)),
+      PaperSize("A5 (148 x 210 mm)",                       MM(148),    MM(210)),
+      PaperSize("A6 (105 x 148 mm)",                       MM(105),    MM(148)),
+      PaperSize("A7 (74 x 105 mm)"",                       MM(74),     MM(105)),
```
Comment (lasconic) [inline libmscore/page.cpp]: You can see there is one " that shouldn't be there. The syntax coloring on github chokes on it

**Comment by lasconic** [comment]
@Jojo-Schmitz  no that will not work.

**Comment by Jojo-Schmitz** [comment]
Well,  `PaperSize(QT_TRANSLATE_NOOP("paperSizes", "Concert Part") " (9 x 12 in)"),  9, 12)` (i.e. without the `+`) does compile

**Comment by lasconic** [comment]
Does it run with the intended result?

**Comment by Jojo-Schmitz** [comment]
It displays alright, it doesn'tsShow a translated string, but it isn't translated yet, so it can't

**Comment by ZackTheCardshark** [comment]
Trying @lasconic's approach for the moment.

File: libmscore/page.cpp [hunk 9]
```diff
@@ -36,38 +36,52 @@ namespace Ms {
...
+      PaperSize(QT_TRANSLATE_NOOP("paperSizes","Concert Part (9 x 12 in)"),
+                9,          12),
+      //: Do not translate "in".
+      PaperSize(QT_TRANSLATE_NOOP("paperSizes","Flip Folder (6.75 x 5.25 in)"),
+                6.75,       5.25),
+      //: Do not translate "in".
+      PaperSize(QT_TRANSLATE_NOOP("paperSizes","Choral Octavo (6.75 x 10.5 in)"),
+                6.75,       10.5),
+      //: Do not translate "in".
+      PaperSize(QT_TRANSLATE_NOOP("paperSizes","Hymn (5.75 x 8.25 in)"),
```
Comment (Jojo-Schmitz) [inline libmscore/page.cpp]: it is not only the 'in' that isn't to be translated, more important is not to translate the numbers

**Comment by Jojo-Schmitz** [comment]
Not sure my method can work. I think because the combined string won't be found in the translations, it would not get replaced with a translation.

File: libmscore/page.cpp [hunk 10]
```diff
@@ -36,38 +36,52 @@ namespace Ms {
...
-      PaperSize("Ledger",    MM(432),  MM(279)),
-      PaperSize("Tabloid",   MM(279),  MM(432)),
-      PaperSize(0,           MM(1),    MM(1))   // mark end of list
+      PaperSize(QT_TRANSLATE_NOOP("paperSizes","Custom"),   MM(1),      MM(1)),
+      PaperSize("Letter (8.5 x 11 in)",                     8.5,        11),
+      PaperSize("Legal (8.5 x 14 in)",                      8.5,        14),
+      PaperSize("Tabloid (11 x 17 in)",                     11,         17),
+      PaperSize("A4 (210 x 297 mm)",                        MM(210),    MM(297)),
+      PaperSize("B5 (176 x 250 mm)",                        MM(176),    MM(250)),
+      PaperSize("Statement (5.5 x 8.5)",                    5.5,        8.25),
```
Comment (shoogle) [inline libmscore/page.cpp]: Size "Statement" is missing units?

Comment (Jojo-Schmitz) [inline libmscore/page.cpp]: and it is obviously meant to be in(ch)

Comment (Jojo-Schmitz) [inline libmscore/page.cpp]: A question is whether to 'spell' the inch decimals (and only those) as fractions, ¼, ½, ¾, as I think this is pretty common with inches? Looks nicer to me.
These are 8-bis ASCII chars, could also use their Unicode counterparts \u00bc, \00bd, \u00be.

Comment (ZackTheCardshark) [inline libmscore/page.cpp]: I think the decimal is a lot more standard; that's what I see in print dialogs and online lists of paper sizes. Thanks for spotting the missing "in."

**Comment by ZackTheCardshark** [comment]
Missing "in" added, rebased and squashed.

**Comment by lasconic** [comment]
I have one more concern about this PR... 1.3 import would use this array via the getPaperSize(name). I will add another commit to make it clear that we remove this and we will need to take care of it if we make a 1.3 import.

**Comment by lasconic** [comment]
It also means that this PR doesn't go into 2.0.4

**Comment by Jojo-Schmitz** [comment]
can't we just compare the leftmost part of the names to get the matching done?

**Pull request merged**